### PR TITLE
Docs BDD intro, some useful links

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Docs BDD intro, some useful links

## Type of change

- Documentation update

## Testing steps

N/A

## Checklist
* [ ] Pylint passes for Python sources
* [ ] sources has been pre-processed by Black
* [ ] updated documentation wherever necessary

### CODE REVIEW

**Review by dpensi — APPROVED** [review]

File: docs/BDD_intro.md [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+---
+
+## Why BDD?
+
+* Today IT systems are usually very complex
+    - quite hard to get the overall picture
+    - overall function vs implementation details
+* Basic/typical problems
+    - bug fixes are very expensive in the later stages of the development
+    - customers don't like features^W bugs
```
Comment (Bee-lee) [inline docs/BDD_intro.md:45]: typo?

Comment (tisnik) [reply docs/BDD_intro.md:45]: well (bad) joke

**Review by Bee-lee — APPROVED** [review]
LGTM, nice overview

File: docs/BDD_intro.md (hunk 1 above)
Comment (Bee-lee) [reply docs/BDD_intro.md:45]: I mean the ^W

**Review by juandspy — APPROVED** [review]
nice explanation actually

**Pull request merged**